packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/DiagramRenderer.tsx

```diff
@@ -109,9 +108,13 @@ export const DiagramRenderer = ({
   }, [diagramRefreshedEventPayload, diagramDescription]);
 
   const { updateSelectionOnNodesChange, updateSelectionOnEdgesChange } = useDiagramSelection(selection, setSelection);
+  const { onBorderChange } = useBorderChange();
+  const { onHandleChange } = useHandleChange();
+  const { onPositionChange } = usePositionChange();
 
   const handleNodesChange: OnNodesChange = (changes: NodeChange[]) => {
-    onNodesChange(onBorderChange(onHandleChange(changes)));
+    const nodeChanges = onBorderChange(onPositionChange(onHandleChange(changes)));
+    onNodesChange(nodeChanges);
```

Comment:
I'm not sure what the behavior will be here, we will have to talk about that tomorrow. Is that deterministic?